docs/serializers.md

```diff
@@ -377,6 +378,71 @@ Both `Color` properties are serialized as strings.
 
 <!--- TEST -->
 
+### Delegating serializers
+
+In the previous example, we represented the `Color` class as string.
+String is considered to be a primitive type, therefore we used `PrimitiveClassDescriptor` and specialized `encodeString` method. 
+Now let's see what our actions would be if we have to serialize `Color` as other non-primitive type, let's say `IntArray`.
+
+An implementation of [KSerializer] for our original `Color` class is going to perform a conversion between
+`Color` and `IntArray`, but delegate the actual serialization logic to the `IntArraySerializer`
+using [encodeSerializableValue][Encoder.encodeSerializableValue] and
+[decodeSerializableValue][Decoder.decodeSerializableValue].
+
+```kotlin
+import kotlinx.serialization.builtins.IntArraySerializer
+
+class ColorIntArraySerializer : KSerializer<Color> {
+    override val descriptor = SerialDescriptor("Color", IntArraySerializer().descriptor)
+
+    override fun serialize(encoder: Encoder, value: Color) {
+      val data = intArrayOf(
+        (value.rgb shr 16) and 0xFF,
```

Comment:
indents are incorrect, should be 4 spaces (to be aligned with Kotlin and our own code style)